### PR TITLE
feat(notification): Implement Post Endpoint for Notification Settings

### LINKED ISSUE #141: [FEAT] Implement Backend for Notification Settings
## Description
Develop and implement the backend components necessary to manage user notification preferences. This includes creating the data models, database schema, API endpoints, and ensuring integration with existing services.
## Purpose:
 To provide a robust and scalable backend solution for storing and managing user notification settings, facilitating efficient data handling and integration with other backend services.

## Requirements:

 1. Design and create a database schema for storing notification preferences.
 2. Develop API endpoints for managing notification settings.
 3. Integrate the notification settings feature with existing user services.
 4. Add authentication to ensure only authenticated users can access the endpoints.
 
## Expected Outcome:

 A robust and scalable data model that accurately stores, retrieves, and manages user notification settings, integrated A comprehensive backend solution that supports the storage, retrieval, and management of user notification settings, integrated seamlessly with the current system architecture.

### Checklist:

- [x]  Define and implement the database schema for notification settings.

- [x]  Create and test database migration scripts.
  
- [x]  Develop and test database migrations scripts to apply the new schema.
  
- [x]   Implement secure and validated API endpoints for CRUD operations on notification settings.
  
- [x]  Add JWT authentication to the POST and GET endpoints for notification settings.
  
- [x]   Integrate the notification settings feature with existing user services.
  
- [x]  Write unit and integration tests for the new backend components.
  
- [x]   Update relevant technical documentation to reflect the new backend features and their integration points.
  
### API Endpoints
POST ```/api/v1/settings/notification-settings``` [PROTECTED]

- **Request:**
```json
{
  "user_id": "user123",
  "mobile_push_notifications": true,
  "activity_workspace_email": false,
  "email_notifications": true,
  "email_digests": true,
  "announcements_update_emails": true,
  "activity_workspace_slack": false,
  "slack_notifications": true,
  "announcements_update_slack": false
}
```
**Response**
```json
  {
    "status": "success",
    "data": {
         "id": 1,
          "user_id": "user123",
          "mobile_push_notifications": true,
          "activity_workspace_email": false,
          "email_notifications": true,
          "email_digests": true,
          "announcements_update_emails": true,
          "activity_workspace_slack": false,
          "slack_notifications": true,
          "announcements_update_slack": false
    }
}
```
Possible Status Code:

200 OK: User is authenticated and settings are updated successfully.
401 Unauthorized: JWT token is missing or invalid.
400 Bad Request: Request payload is invalid.

Get Notification Settings:
GET ```/api/v1/settings/notification-settings/:user_id``` [PROTECTED]
- **Response:**
```json
  {
    "status": "success",
    "data": {
         "id": 1,
         "user_id": "user123",
          "mobile_push_notifications": true,
          "activity_workspace_email": false,
          "email_notifications": true,
          "email_digests": true,
          "announcements_update_emails": true,
          "activity_workspace_slack": false,
          "slack_notifications": true,
          "announcements_update_slack": false
    }
}
```

Possible Status Code:

200 OK: User is authenticated and settings are updated successfully.
401 Unauthorized: JWT token is missing or invalid.
400 Bad Request: Request payload is invalid.

**REFERENCE:** [**APPROVED ISSUE**](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/20)

### PR DESCRIPTION
**Pull request by Reliable25**
# Description

**Closes #141**

# Changes proposed

## What were you told to do?

Develop and implement the backend components necessary to manage user notification preferences. This includes creating the data models, database schema, API endpoints, and ensuring integration with existing services.
## What did you do?

Design and create a database schema for storing notification preferences.
Develop API endpoints for managing notification settings.
Integrate the notification settings feature with existing user services.
Add authentication to ensure only authenticated users can access the endpoints.

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos

### CODE REVIEW

File: src/Hng.Application/Features/Notifications/Commands/CreateNotificationCommand.cs
Comment (Godhanded) [inline src/Hng.Application/Features/Notifications/Commands/CreateNotificationCommand.cs:1]: This file doest seem right, its looking like a dto but this file is for a command

**Review by Godhanded — CHANGES_REQUESTED** [review]

File: src/Hng.Application/Features/Notifications/Commands/CreateNotificationCommand.cs
Comment (Reliable25) [reply src/Hng.Application/Features/Notifications/Commands/CreateNotificationCommand.cs:1]: have done it, pls check @Godhanded

File: src/Hng.Application.Test/Features/UserManagement/GoogleLoginCommandHandlerTests.cs [hunk 1]
```diff
@@ -91,7 +91,7 @@ public async Task Handle_ShouldRegisterNewUser_WhenUserDoesNotExist()
             Assert.NotNull(result);
             Assert.Equal("Registration successful, user logged in.", result.Message);
             Assert.NotNull(result.AccessToken);
-            //Assert.Equal("newuser@example.com", result.Data.Email);
+            Assert.Equal("newuser@example.com", result.Data.Email);
```
Comment (Godhanded) [inline src/Hng.Application.Test/Features/UserManagement/GoogleLoginCommandHandlerTests.cs:94]: Please comment back this line,

File: src/Hng.Application.Test/Features/UserManagement/GetUsersQueryShould.cs [hunk 2]
```diff
@@ -60,8 +60,8 @@ public async Task ReturnMultipleUsers()
 
                 Assert.Equal(expected.Id, result.Id);
                 Assert.Equal(expected.Email, result.Email);
-                Assert.Contains(expected.FirstName, result.FullName);
-                Assert.Contains(expected.LastName, result.FullName);
+                Assert.Contains(expected.FirstName, result.Name);
+                Assert.Contains(expected.LastName, result.Name);
```
Comment (Godhanded) [inline src/Hng.Application.Test/Features/UserManagement/GetUsersQueryShould.cs:64]: result Full name ,no longer name

File: src/Hng.Application.Test/Features/UserManagement/GetUserByIdQueryShould.cs [hunk 3]
```diff
@@ -49,8 +49,8 @@ public async Task ReturnASingleUser()
             // Assert
             Assert.Equal(expected.Id, result.Id);
             Assert.Equal(expected.Email, result.Email);
-            Assert.Contains(expected.FirstName, result.FullName);
-            Assert.Contains(expected.LastName, result.FullName);
+            Assert.Contains(expected.FirstName, result.Name);
+            Assert.Contains(expected.LastName, result.Name);
```
Comment (Godhanded) [inline src/Hng.Application.Test/Features/UserManagement/GetUserByIdQueryShould.cs:53]: result.FullName no longer result.name

File: src/Hng.Application.Test/Features/UserManagement/GoogleLoginCommandHandlerTests.cs [hunk 4]
```diff
@@ -139,7 +139,7 @@ public async Task Handle_ShouldLoginExistingUser_WhenUserExists()
             Assert.NotNull(result);
             Assert.Equal("Login successful", result.Message);
             Assert.NotNull(result.AccessToken);
-            //Assert.Equal("existinguser@example.com", result.Data.user.Email);
+            Assert.Equal("existinguser@example.com", result.Data.Email);
```
Comment (Godhanded) [inline src/Hng.Application.Test/Features/UserManagement/GoogleLoginCommandHandlerTests.cs:142]: Please comment this line again

File: src/Hng.Application.Test/Features/UserManagement/LoginUserCommandShould.cs [hunk 5]
```diff
@@ -76,7 +76,7 @@ public async Task ReturnLoginResponseDtoForValidCredentials()
             Assert.Equal("Login successful", result.Message);
             Assert.Equal("token", result.AccessToken);
             Assert.NotNull(result.Data);
-            //Assert.Equal(_user.Email, result.Data.Email);
+            Assert.Equal(_user.Email, result.Data.Email);
```
Comment (Godhanded) [inline src/Hng.Application.Test/Features/UserManagement/LoginUserCommandShould.cs:79]: Comment back this line

**Review by Godhanded — CHANGES_REQUESTED** [review]
this pr is reverting many changes that have already been merged back to before they where, try cloning the repo again and pull to have all the latest changes then add your features and open a pr again

**Pull request closed without merging**